docker/entrypoint.sh

```diff
@@ -1,128 +1,292 @@
 #!/usr/bin/env bash
```

Comment:
Optional: some developers have asked for a short summary of what each CI / develops file does at the top of the file. One sentence is enough.

Reply:
Fixed here [chore: add comments to main parts of the file](https://github.com/ZcashFoundation/zebra/pull/7660/commits/9f7a92cd4a0c3e1839df30f9f3416e3c3dbb3e81)